frontend/server/src/DAO/Clarifications.php

```diff
@@ -106,6 +100,28 @@ final public static function getProblemsetClarifications(
             $params
         );
 
+        // If we didn't get an offset, we know the total number of rows
+        // already, no need to query the database for it.
+        $totalRows = count($clarifications);
+        if (!is_null($offset)) {
```

Comment:
nit: una micro-optimización más.

```suggestion
        if (!is_null($offset) && $offset != 0) {
```